### ISSUE TITLE
z/OS squad creation

### ISSUE THREAD

**Issue by 1000TurquoisePogs** [zowe/community#1460]
# Name of squad
  *  z/OS squad (z/OS interface squad)
  
# Squad description
A squad for development of z/OS native code used by Zowe, such as ZSS and ZIS

## Origin and history
ZSS & ZIS were contributed during the founding of Zowe as a way to build network APIs for managing z/OS data securely, so that portable servers and clients can work with z/OS in a vendor neutral and open source way.

ZIS and ZSS have some libraries in common, but both can be run standalone. Like many other parts of Zowe, there are conditional dependencies so that when more of Zowe is used together, more features are accessible. However, it's important to note that despite this code currently living in the same squad as the app-server and Desktop, ZIS and ZSS have no dependency on the app-server or Desktop, and the app-server and Desktop can be run without ZSS or ZIS as well. The only commonality is mutually understandable configuration files (as seen in all zowe servers in v2) and plugin metadata. The code and skills required for ZSS and ZIS are entirely different from that of the rest of the WebUI squad, with ZIS being written in Metal C and ZSS being written in LE C.

## What it does, why its valuable
  *  We aim to be a squad which can be consulted and likely the source of new Zowe enhancements that need z/OS native code
  *  A group of mainframe experts that can help to guard against insecure coding, and promote enterprise-ready mainframe services and interfaces
  *  Boost ZSS, ZIS development by making visible a place to attract new talent and educate them
      * Also serves as a visible place to be educated on general principles of secure z/os application development
  *  Supply interfaces to z/OS capbilities that the rest of Zowe or other projects/products want to leverage, such as passtickets, filesystem management, RBAC
      * Does so in a plugable way for extensibility in and outside of the community
  *  Has a distinctly different value from the Desktop and app-server, which depend upon such low-level services but create value by client-side UI modernization instead.
      * Developers of the z/OS code and the Desktop code may have different but both valuable opinions on community design and decision making, so these voices should be heard seperately
  *  The servers in question are independent of other servers of Zowe, and sometimes are seen in the field by themselves.
     *  Both ZSS and ZIS can exist in the absense of other components of Zowe, and therefore should be tracked and managed independent of those other components.
     *  ZIS is a standalone server. 
         * Today, it is sometimes repackaged for use in products outside of Zowe following the terms of our open source license.
     *  ZSS is a standalone server, it is often seen as using ZIS but can run without it.
         * We've collaborated with development groups looking to use this on it's own, too.
     *  app-server and apiml are independent of zss+zis. They are more capable in its presense, but they collaborate at an API level
  
# Statement on alignment with Zowe mission and principles
  * "Open first" - ZIS and ZSS greatly contributes to z/OS application development possible for low-level programmers by providing working examples of how to interface with z/OS structures and serving as a base for others to make native z/OS projects in open source communities
  
  * "Security is paramount" - ZIS runs APF, so it is at the same level of permission as essential, highly privileged components of z/OS such as VTAM. Therefore, this group must take security seriously. 
      * Examples: Cross-memory framework which provides SAF mediated syscalls/call-gate from a webserver
  
  * "Modern and effective interfaces" - ZIS and ZSS used together present a secure way for more portable, less privileged services to access mainframe data. ZIS+ZSS are Zowe's "bridge" to z/OS and present it through modern interfaces: HTTPS REST APIs, Websockets, more. As a Zowe component, the community can provide such interfaces on our own timeline, independent of vendors. 
      * Examples: Dynamic Linking, 64bit HTTPS Server, REST and Websocket APIs
  
  * "Hybrid" - ZIS and ZSS, and servers which call ZSS together operate in a hybrid environment in which servers calling ZSS do not need to be on z/os, and zss+zis will handle all of the mainframe-specific data on behalf of other servers
  
  * "Stakeholder-Centric" - ZIS+ZSS have been a part of every multi-squad initiative in Zowe server enhancements including MFA, SSO, HA/FT, but also have had enhancements specific to them based on user feedback such as Dynamic Linking & 64 bit. This group will also be continuing to collaborate with the server infrastructure group on initiatives such as PARMLIB support and schema support
  
  * "Vendor neutral" - ZIS+ZSS are often a solution to the vendor neutral requirements of Zowe as a whole, because if certain z/OS features of today are only found in certain vendor products, the only way to get such support on our own terms may be to write native code, which would be found in ZIS+ZSS.
  
  * "Minimal viable governance" - We believe this squad has a right to exist because we see no reasons why it shouldn't, and many of the above reasons why it should.
  
# Ambassador from Zowe (helps mentor projects) - Preferrably TSC or ZAC member
  *  Irek Fakrhudinov, Sean Grady
  
# External dependencies (including licenses)
  *  None
  
# Initial committers
  * Irek Fakrhudinov (Zowe committer since founding)
  * Leonty Chudinov (Zowe committer since founding)
  * Joe Devlin (Zowe committer since founding)
  * Joe Pun (Joined around January 2021)

# Infrastructure requests (CI / OMP Cluster)
  * None, existing infrastructure already in place in Zowe

# Communication channels (Slack, irc, mailing lists) - Zowe has Slack channels, Mailing lists and forums, let us know if you already have channels we should be aware of. 
  * None, existing channels and mailing lists are good.

# Preferred stage (incubation, active)
  * Active

**Comment by Joe-Winchester** [comment]
I like this proposal a lot.  Now that ZSS has TLS there is a lot of opportunity for adding more native services that other clients might want access to through REST.  I agree that having this expertize within Zowe and not always having to rely on z/OSMF or other native components to provide functionality is a great thing for Zowe.
I'm not sure about the name (sorry to bikeshed).  I'm not sure "z/OS squad" is the right name.  Maybe "z/OS Base", or "z/OS Kernel" ?

**Comment by 1000TurquoisePogs** [comment]
Yeah - it's z/OS something. I suggested "z/OS" because i figure it's not just the code, its also the squad in which z/os experts will be found, and which would be most likely to voice z/os concerns or vote on readiness of z/os things.

**Comment by armstro** [comment]
When in doubt make something up....like.... "Zowe z/OS Anchor" ...if we do more with containers in cloud or off z/OS then I _assume_ this would still reside on z/OS. {8-)

**Comment by Joe-Winchester** [comment]
- Does the Zowe launcher belong in the remit of this squad, or should it belong in the systems squad and be part of the zowe-install-packaging repo ?
- Likewise for the config manager ?

My worry is that "z/OS squad" is too big a remit,  and also being "everything written in C" is also wrong because then this squad would own and cover the launcher and the config manager, which conceptually are in the remit of the install/packager

**Comment by 1000TurquoisePogs** [comment]
low-level needs for server infrastructure like install/package/run seem to exist between the two squads. Perhaps expertise found in the z/OS squad would be required to write things to be consumed in the infrastructure. It's up to these two squads to cooperate to determine the best place for them to be maintained. This would not be the first or last time we've had long term cross-squad collaboration.
